### PR TITLE
[WEB-223] WIP: Feature: Add donation to Fuel Rats website

### PR DESCRIPTION
**Pull request by itssimple**
## Changes
- Modifies the main menu to point to `/donate` instead of separate domain

## Features
- Adds page to handle donations
  - View for selecting payment type, currency and amount
  - View for confirming your donation and submitting it
  - Webhook handler for receiving confirmation from Stripe

### CODE REVIEW

File: client/components/Donate/DonateSelectPaymentType.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import ValidatedCurrencySelect from '../ValidatedCurrencySelect'
+
+
+const INVALID_CURRENCY_MESSAGE = 'Currency is Required'
+const initialState = {
```
Comment (UncleClapton) [inline client/components/Donate/DonateSelectPaymentType.js]: unless you need to reinitialize component state, just define state members within the class property.

File: client/components/Donate/DonateSelectPaymentType.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    currency: INVALID_CURRENCY_MESSAGE,
+  },
+}
+
+class DonateSelectPaymentType extends React.Component {
+  state = {
+    ...initialState,
+  }
+
+  static noop () {
```
Comment (UncleClapton) [inline client/components/Donate/DonateSelectPaymentType.js]: nice.

**Review by UncleClapton — CHANGES_REQUESTED** [review]

File: client/components/Donate/DonateSelectPaymentType.js (hunk 1 above)
Comment (itssimple) [reply client/components/Donate/DonateSelectPaymentType.js]: I have fixed this. :D

File: client/components/Donate/DonateSelectPaymentType.js (hunk 2 above)
Comment (itssimple) [reply client/components/Donate/DonateSelectPaymentType.js]: This as well!

File: client/components/Donate/DonateSelectPaymentType.js (hunk 1 above)
Comment (itssimple) [reply client/components/Donate/DonateSelectPaymentType.js]: I like trains

File: client/components/Donate/DonateSelectPaymentType.js (hunk 2 above)
Comment (itssimple) [reply client/components/Donate/DonateSelectPaymentType.js]: I still like trains

**Comment by itssimple** [comment]
And this is a real comment, not a reply on another reviewed comment.

**Pull request closed without merging**